### PR TITLE
Added Movies model

### PR DESCRIPTION
**Pull request by milanes**

### CODE REVIEW

File: test/unitTestsMovies.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+const test = require("tape");
```
Comment (osedoe) [inline test/unitTestsMovies.js]: If you have a controller.js file, the test file tends to be controller.test.js

File: models/Movies.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Es una clase de Javascript con un constructor que toma 3 cadenas y las almacena.
+ * Esto actúa como el esquema para los datos en MongoDB.
+ */
```
[the remark this answers is not in the document]
Comment (milanes) [reply models/Movies.js]: Now it's done.

File: routes/routes.js [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
 const Document = require("../models/Document");
+const Movie = require("../models/Movies");
 const router = express.Router();
 
-//Obtiene todos los documentos en la colección - /documents/all
+/**
+ * Documents
+ * Obtiene todos los documentos en la colección - /documents/all
+ */
 router.get("/documents/all", (req, res, next) => {
```
[the remark this answers is not in the document]
Comment (milanes) [reply routes/routes.js]: In the new push you can see the change of my branch movies-config

File: test/unitTestsMovies.js (hunk 1 above)
Comment (milanes) [reply test/unitTestsMovies.js]: Now the test works, but i need learn how move to controller.test.js

**Comment by milanes** [comment]
Added Controller, new Router and delete Documents.js files and its dependecies. I need your check for the merge!

File: controllers/movieController.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+exports.movie_details = function(req, res, next) {
+  req.app.locals.db.collection("movies").findOne(
+    {
+      _id: req.params.id
+    },
+    (err, result) => {
+      if (err) {
+        res.status(400).send({ error: err });
+      }
+      if (result === undefined) {
```
Comment (osedoe) [inline controllers/movieController.js]: You can compare against undefined  and null at the same time like this:
```
if (result) {
  res.status...
}

File: controllers/movieController.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        res.status(400).send({ error: "No movie matching that id was found" });
+      } else {
+        res.status(200).send(result);
+      }
+    }
+  );
+};
+
+// Add a new movie to database
+exports.movie_add = function(req, res, next) {
```
Comment (osedoe) [inline controllers/movieController.js]: Why not:
```
const { title, year, author, body} = req.body;
const newMovie = new Movie(title, year, author, body);

File: .editorconfig [hunk 6]
```diff
@@ -0,0 +1,12 @@
+root = true
```
Comment (osedoe) [inline .editorconfig]: I'd say that you need to add this file to your `.gitignore`, but ask Raul first.

File: controllers/movieController.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
+const Movie = require("../models/Movies");
+
+// List all movies from MongoDB database
+exports.movies_list = function(req, res, next) {
```
Comment (osedoe) [inline controllers/movieController.js]: JavaScript naming convention uses camelCase to name functions and variables, so I'd put `moviesList` and stick with that. Only use the lower score for constants, like:
```
const NUMBER_OF_MOVIES = 3;
```
PS. Keep in mind that not all `const` are CONSTANTS.

File: controllers/movieController.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+// Add a new movie to database
+exports.movie_add = function(req, res, next) {
+  const newMovie = new Movie(
+    req.body.title,
+    req.body.year,
+    req.body.author,
+    req.body.body
+  );
+  req.app.locals.db.collection("movies").insertOne(
```
Comment (osedoe) [inline controllers/movieController.js]: I'm not a big fan of putting the callback of a function underneath, I didn't get it at first.
I'd write it as:
```
... .insertOne({ newMovie }, (err, res) => {
  // code
}
```
Try to use the state space of the screen to convey for information in each line and separate concerns vertically. But as I said, it's just a matter of preference.

File: .editorconfig (hunk 6 above)
Comment (milanes) [reply .editorconfig]: Raúl say that this file will be excluded of the .gitignore. That help to others developer...  ;)

File: controllers/movieController.js (hunk 7 above)
Comment (milanes) [reply controllers/movieController.js]: It's done.  TY

File: controllers/movieController.js (hunk 5 above)
Comment (milanes) [reply controllers/movieController.js]: Solved!

File: controllers/movieController.js (hunk 8 above)
Comment (milanes) [reply controllers/movieController.js]: Solved!

**Comment by milanes** [comment]
Merge is possible now?

**Comment by osedoe** [comment]
La PR esta aprobada, pero te dejo que la mergees tu @milanes  ;)

**Pull request merged**